libs/providers/launchdarkly-client/src/lib/launchdarkly-client-provider.ts

```diff
@@ -34,73 +38,110 @@ export class LaunchDarklyClientProvider implements Provider {
     name: 'launchdarkly-client-provider',
   };
 
+  private readonly ldOptions: LDOptions | undefined;
   private readonly logger: Logger;
+  private _client?: LDClient;
+
+  public events = new OpenFeatureEventEmitter();
+
+  /*
+   * implement status field/accessor
+   * https://openfeature.dev/specification/sections/providers#requirement-242
+   * */
+  private _status: ProviderStatus = ProviderStatus.NOT_READY;
 
-  constructor(private client: LDClient, options: LaunchDarklyProviderOptions = {}) {
-    if (options.logger) {
-      this.logger = options.logger;
+  set status(status: ProviderStatus) {
+    this._status = status;
+  }
+
+  get status() {
+    return this._status;
+  }
+
+  constructor(private readonly envKey: string, { logger, ...ldOptions }: LaunchDarklyProviderOptions) {
+    if (logger) {
+      this.logger = logger;
     } else {
       this.logger = basicLogger({ level: 'info' });
     }
+    this.ldOptions = { ...ldOptions, logger: this.logger };
+
+  }
+
+  private get client(): LDClient {
+    if (!this._client) {
+      throw new GeneralError('Provider is not initialized');
+    }
+    return this._client;
   }
 
-  async initialize(context: EvaluationContext): Promise<void> {
-    const newContext = this.translateContext(context);
-    const oldContext = this.client.getContext();
-    const ignoreAnonymous = newContext.anonymous && oldContext.anonymous && !newContext.key
-    if(!ignoreAnonymous && !isEqual(newContext, oldContext)) {
-       await this.client.identify(newContext);
+  async initialize(): Promise<void> {
+    /* for simplicity always initialize with anonymous by default*/
+
+    this._client = initialize(this.envKey, { anonymous: true }, this.ldOptions);
+
+    /*
+     * set event listeners to propagate ld events to open feature provider,
+     * use LD property streaming, since when enabled you are opting in for straming updates
+     * */
+    if (this.ldOptions?.streaming) {
+      this.setListeners();
```

Comment:
This is probably fine, but the SDK itself would typically also start streaming if anyone listened for updates. So, this is kind of an indirect way of opting in.

Because of that I think we should add a note to the readme indicating that events will only be emitted if streaming is enabled.

Reply:
> This is probably fine, but the SDK itself would typically also start streaming if anyone listened for updates. So, this is kind of an indirect way of opting in.
> 
> Because of that I think we should add a note to the readme indicating that events will only be emitted if streaming is enabled.

In that case, we should probably add a note to the readme as you say, AND extend/override the jsdoc on the configuration object (I think you can add a link to the overridden jsdoc as well).

cc @sago2k8

Reply:
I did added some more context in the README but nothing really in the jsdoc, do you happen to have an example on how to override the jsdoc properly?